src/libraries/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -270,71 +273,103 @@ private void CleanupEventStream()
                 if (eventStream != null)
                 {
                     _cancellationRegistration.Unregister();
-                    try
-                    {
-                        // When we get here, we've requested to stop so cleanup the EventStream and unschedule from the RunLoop
-                        Interop.EventStream.FSEventStreamStop(eventStream);
```

Comment:
FSEventStreamStop does not throw any exceptions. This try/finally does not look necessary. It may be a left over from some thread abort hardening.